DependencyInjection/HttplugPluginClientFactoryCompilerPass.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auxmoney\OpentracingHttplugBundle\DependencyInjection;
+
+use Auxmoney\OpentracingHttplugBundle\Factory\DecoratedPluginClientFactory;
+use Http\Client\Common\PluginClient;
+use Http\Client\Common\PluginClientFactory;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+class HttplugPluginClientFactoryCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        foreach ($container->getDefinitions() as $definition) {
+            if ($definition->getClass() === PluginClient::class) {
+                $callable = $definition->getFactory();
+
+                if ($callable[0] instanceof Reference && (string)$callable[0] === PluginClientFactory::class) {
```

Comment:
Doesn't this break when `$callable` is `null` or `string`, as it could be?

Reply:
Definitely! thanks